crates/holochain/src/core/workflow/publish_dht_ops_workflow/publish_query.rs

```diff
@@ -172,6 +203,88 @@ mod tests {
         assert_eq!(expected.results.len() + 1, num_to_publish);
     }
 
+    #[tokio::test(flavor = "multi_thread")]
+    async fn publish_query_includes_warrants() {
+        holochain_trace::test_run();
+
+        let agent = fixt!(AgentPubKey);
+        let db = test_authored_db();
+
+        // Insert one chain op and one warrant op into database.
+        let chain_op = create_and_insert_op(
+            &db,
+            Facts {
+                private: false,
+                within_min_period: false,
+                has_required_receipts: false,
+                is_this_agent: true,
+                store_entry: false,
+                withold_publish: false,
+            },
+            &Consistent {
+                this_agent: agent.clone(),
+            },
+        )
+        .await
+        .content;
+        let warrant_op = insert_invalid_op_warrant_op(&db, &agent).content;
+
+        let ops_to_publish = get_ops_to_publish(
+            agent.clone(),
+            &db.to_db(),
+            ConductorTuningParams::default().min_publish_interval(),
+        )
+        .await
+        .unwrap()
+        .into_iter()
+        .map(|(_, _, op)| op)
+        .collect::<Vec<_>>();
+        assert_eq!(ops_to_publish, vec![chain_op, warrant_op]);
+
+        let num_to_publish = db
+            .to_db()
+            .test_read(|txn| num_still_needing_publish(txn, agent).unwrap());
+        assert_eq!(num_to_publish, 2);
+    }
+
+    fn insert_invalid_op_warrant_op(
```

Comment:
This is a little confusing to read, perhaps `insert_invalid_chain_op_warrant_op` helps a little?